packages/downloads/src/portal-request-dataset-export.ts

```diff
@@ -0,0 +1,103 @@
+import { RemoteServerError } from './remote-server-error';
+import { urlBuilder } from './utils';
+import { IPortalDatasetExportRequestParameters } from './portal-dataset-export-request-parameters';
+
+
+/**
+ * ```js
+ * import { portalRequestDatasetExport } from "@esri/hub-downloads";
+ * 
+ * const params = {
+ *   host: 'https://hub.arcgis.com/portal,
+ *   datasetId: 'abcdef0123456789abcdef0123456789_0',
+ *   format: 'CSV',
+ *   username: 'user',
+ *   token: '123'
+ * };
+ * portalRequestDatasetExport(params)
+ *   .then(response => {
+ *     // {
+ *     //   downloadId: 'abcdef0123456789abcdef0123456789_0::CSV',
+ *     // }
+ *   });
+ * ```
+ * Request an export of a dataset to a particular file format.
+ * @param params - parameters defining a dataset export job
+ */
+export function portalRequestDatasetExport (params: IPortalDatasetExportRequestParameters) {
+  const {
+    host,
+    datasetId,
+    format,
+    username,
+    token
+  } = params;
+  const [itemId, layerId = 0] = datasetId.split('_');
+  const exportParameters = {
+    layers: [{ id: layerId }]
+  }
+  return new Promise((resolve, reject) => {
+    exportItem({
+        portal: host,
+        itemId,
+        exportFormat: format,
+        exportParameters,
+        username,
+        token
+      })
+    .then(resolve)
+    .catch(reject);
+  });
+}
+
+interface IExportItemResult {
+  downloadId: string;
+  jobId: string;
+  exportCreated: number;
+  status: string;
+}
+
+// Hoist to REST JS
+function exportItem (params: any): Promise<IExportItemResult> {
+  const {
```

Comment:
One thought on this - should we fetch the item and do some type checking on it before we fire off the request to export it? That way we can return a descriptive error, vs whatever the API returns.